clustered/client/src/test/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntityTest.java

```diff
@@ -0,0 +1,148 @@
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal.store;
+
+import org.ehcache.clustered.common.internal.ServerStoreConfiguration;
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityMessage;
+import org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse;
+import org.ehcache.clustered.common.internal.messages.EhcacheResponseType;
+import org.junit.Test;
+import org.mockito.stubbing.Answer;
+import org.terracotta.entity.EndpointDelegate;
+import org.terracotta.entity.EntityClientEndpoint;
+import org.terracotta.entity.InvocationBuilder;
+import org.terracotta.entity.InvokeFuture;
+import org.terracotta.entity.MessageCodecException;
+import org.terracotta.exception.EntityException;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests that the cluster tier client entity fires any {@link ClusterTierClientEntity.ResponseListener} only after
+ * the {@link ClusterTierClientEntity#validate(ServerStoreConfiguration)} method is called.
+ */
+public class SimpleClusterTierClientEntityTest {
+  private static final int MAX_TEST_WAIT_TIME_SECONDS = 30;
+
+  @Test
+  public void testFireWithInit() throws Exception {
+    MockEndpointBuilder eb = new MockEndpointBuilder();
+    SimpleClusterTierClientEntity entity = new SimpleClusterTierClientEntity(eb.mockEndpoint);
+    AtomicBoolean responseRcvd = new AtomicBoolean(false);
+    entity.addResponseListener(MockedEntityResponse.class, response -> responseRcvd.set(true));
+    entity.validate(mock(ServerStoreConfiguration.class));
+    eb.mockInvocationBuilder.message(mock(EhcacheEntityMessage.class));
+    assertTrue(responseRcvd.get());
+  }
+
+  @Test
+  public void testFireWithDelayedInit() throws Exception {
+    MockEndpointBuilder eb = new MockEndpointBuilder();
+    SimpleClusterTierClientEntity entity = new SimpleClusterTierClientEntity(eb.mockEndpoint);
+    AtomicBoolean responseRcvd = new AtomicBoolean(false);
+    entity.addResponseListener(MockedEntityResponse.class, response -> responseRcvd.set(true));
+    CountDownLatch beforeLatch = new CountDownLatch(1);
+    CountDownLatch afterLatch = new CountDownLatch(1);
+    Executors.newFixedThreadPool(1).submit(() -> {
+      beforeLatch.countDown();
+      eb.mockInvocationBuilder.message(mock(EhcacheEntityMessage.class));
+      afterLatch.countDown();
+    });
+    assertTrue(beforeLatch.await(MAX_TEST_WAIT_TIME_SECONDS, TimeUnit.SECONDS));
+    Thread.sleep(100);
```

Comment:
I think instead of the sleep, you want 
```
long end = System.currentMillis() + 10_000;
while(responseRcvd.get()) {
  if(System.currentMillis() > end) {
    fail("Response listener was never called");
    break;
  }
}
```

Reply:
just to get the other thread some time to enter its wait..so that this thread does not complete its init too fast...

Reply:
It is only to force yield?  (then line comment it). I was wondering if there should be a `while(responseRcvd.get())` instead of the assert below.

Reply:
done